test/files/run/matchonstream.scala

```diff
@@ -1,3 +1,3 @@
 object Test extends App{
-  LazyList.from(1) match { case LazyList(1, 2, x @_*) => println("It worked!") }
+  LazyList.from(1) match { case LazyList(1, 2, x @_*) => println(s"It worked! (class: ${x.getClass.getSimpleName})") }
```

Comment:
this change to prevent DCE

Reply:
Does that mean that the `match` expression was not even evaluated because of DCE?

Reply:
I think, based on @smarter's comment, just the `x @_*` (which is `UnapplySeqWrapper#drop`) wasn't evaluated?

Reply:
patmat never generates the code in the first place if the symbol is unused.